### DISCUSSION TITLE
Does it require Go hosting

### DISCUSSION THREAD

**Discussion by ugintl** [JordanKnott/taskcafe#165] [Q&A]
Does it require Go hosting like php hosting for php app?

**Comment by GwynethLlewelyn** [comment]
Not in the same sense, no. To have PHP support in hosting services, you need either to have a built-in interpreter module for PHP (which can be embedded in Apache, for example) or a stand-alone PHP interpreter/compiler which the webserver will talk to whenever it needs to run some PHP code (this is the approach used by nginx and many others, and can _also_ be used with Apache). Both these solutions require your hosting provider to set this environment up for you, assuming you don't have direct access to the system's superuser (root) account.

Instead, to run Taskcafé, you need to have access to the system's command-line interface and the ability to install software there. Having support for Docker containers makes everything much easier. In theory, you don't need any webserver at all (Taskcafé just runs off a port exposed to the Internet); in practice, for security purposes (as well as some reverse proxy/caching), it's customary to run a reverse proxy in front of Taskcafé — the 'outside world' will only know they're talking to a webserver and have no clue what is running behind it.
